### ISSUE TITLE
Native BVH Os Incompability

### ISSUE THREAD

**Issue by empirephoenix** [jMonkeyEngine/jmonkeyengine#254]
Native Bullet:

Generate a Meshcollisionshape on 64bit Windows 7.
Write it out with the native bvh tree.

Load on same Windows works flawless.
Load on linux creates native crashes as apparently the native structure between different OS is not guranteed to be stable.

Proposed Solution:

Upon writing the MeshCollisionShape save the os.name and os.architecture properties into the j3o. When loading check if the architecture matches, if not emit warning to Logger and call createShape(true);
If the j3o is an older one with no saved os.name default to emit a warning and rebuilding always the shape.

Expected Result:
If the j3o does not contain the right native data, the application can continue to work, at an increased loading time for the j3o

Additional Notes:
Removing the native bvh save is in my opinon no options, as it can increase loading time by a factor of roughly 10x while reducing the memory print. This is especially noticeable for large geometries like whole levels. The bullet documentation specifically says to consider saving this native structure.

@normen 
Any objection to this approach? If not i will create a pull request.

**Comment by shadowislord** [comment]
What is bullet's stance on this? How can they recommend persisting the data structure if it is not portable across platforms?

**Comment by empirephoenix** [comment]
I made a thread in their forum 
http://www.bulletphysics.org/Bullet/phpBB3/viewtopic.php?f=9&t=10547

**Comment by empirephoenix** [comment]
Just saying, I'm still on this.
With the answer form the bullet forums, I think I might be able to fix this so it actually works reliable.

**Comment by empirephoenix** [comment]
Ok, after some more time with this, I say lets axe this for the time being. 
Better to have one less feature but better stability.

**Comment by shadowislord** [comment]
@empirephoenix 
Have you looked at the section "Saving a single object, shape or acceleration structure"?
http://bulletphysics.org/mediawiki-1.5.8/index.php/Bullet_binary_serialization

**Comment by empirephoenix** [comment]
Yes I did, but I failed to get it to work. Actually the writing part seems to work fine with that approach.

However the reading part, I did not manage to get btBulletWorldImporter. This might be due to a combination of my missing c++ knowledge combined with out grade based compiling. I did not manage to get that part to work. The importer lies in the extras subproject. It appears this extra part is not at all compiled at the moment, and also not included into the include path.
